syntax = "proto3";

package user;

option go_package="./app/grpc/rpc_pb";

service LoginSignupService {
  rpc FindUserByCredential(FindUserRequest) returns (UserResponse) {}
  rpc CreateUser(CreateRequest) returns (UserResponse) {}
}

message CreateRequest {
  string username = 1 [json_name = "username"];
  string email = 2 [json_name = "email"];
  string first_name = 3 [json_name = "first_name"];
  string last_name = 4 [json_name = "last_name"];
  string password = 5 [json_name = "password"];
}

message FindUserRequest {
  optional string username = 1;
  optional string email = 2;
  string password = 3;
}

message UserResponse {
  string id = 1;
  string username = 2;
  string email = 3;
  string first_name = 4;
  string last_name = 5;
}
